packages/core/src/dispatcher.ts

```diff
@@ -345,21 +345,27 @@ export class Dispatcher {
    */
   async _handleResponseMessage(message: ResponseMessage): Promise<void> {
     const { id: queryId, tips } = message
-    const expectedStreamID = this.messageBus.outstandingQueries.get(queryId)
-    if (expectedStreamID) {
-      const newTip = tips.get(expectedStreamID.toString())
-      if (!newTip) {
-        throw new Error(
-          "Response to query with ID '" +
-            queryId +
-            "' is missing expected new tip for StreamID '" +
-            expectedStreamID +
-            "'"
-        )
-      }
-      this.repository.stateManager.update(expectedStreamID, newTip)
-      this.messageBus.outstandingQueries.delete(queryId)
-      // TODO Iterate over all streams in 'tips' object and process the new tip for each
+    const outstandingQuery = this.messageBus.outstandingQueries.queryMap.get(queryId)
```

Comment:
what if this response message isn't for a query we sent?  Then the outstanding query won't be in the set.  Looks like as written we would wind up throwing an exception on line 352, but it's totally expected behavior that pubsub might contain RESPONSE messages that have nothing to do with this node.  We shouldn't error in that case.  We should probably check if there's no outstanding query for this queryId and early return without error from _handleResponseMessage if so